var/spack/repos/builtin/packages/py-meldmd/package.py

```diff
@@ -1,28 +1,88 @@
-# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
+
 from spack.package import *
 
 
-class PyMeldmd(PythonPackage, CudaPackage):
+class PyMeldmd(CMakePackage, PythonPackage, CudaPackage):
     """MELD is a tool for inferring the structure of
     biomolecules from sparse, ambiguous, or noisy data."""
 
     homepage = "http://meldmd.org/"
     url = "https://github.com/maccallumlab/meld/archive/refs/tags/0.4.20.tar.gz"
 
+    version("0.6.1", sha256="aae8e5bfbdacc1e6de61768a3298314c51575cda477a511e98dc11f5730fd918")
     version("0.4.20", sha256="8c8d2b713f8dc0ecc137d19945b3957e12063c8dda569696e47c8820eeac6c92")
 
+    extends("python")
+
     depends_on("python@3.6:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("amber")
-    depends_on("openmm")
+    depends_on("openmm+cuda")
     depends_on("py-netcdf4", type=("build", "run"))
-    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-numpy@:1.23", type=("build", "run"))
```

Comment:
Can you add a comment directly to the file explaining the upper bound?

Reply:
Sorry, it was numpy.bool that gave me the error, but that was deprecated at the same time as numpy.int:

```
/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/lib/python3.8/site-packages/meld/vault.py:267: FutureWarning: In the future
 `np.bool` will be defined as the corresponding NumPy scalar.  (This may have returned Python scalars in past versions.
  return self._cdf_data_set.variables["positions"][..., stage]
Traceback (most recent call last):
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/bin/launch_remd", line 35, in <module>
    main()
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/bin/launch_remd", line 30, in main
    launch.launch(args.platform, console, args.debug, args.console_log)
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/lib/python3.8/site-packages/meld/remd/launch.py", line 111, in launch
    remd_runner.run(communicator, system_runner, store)
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/lib/python3.8/site-packages/meld/remd/leader.py", line 105, in run
    states = store.load_states(stage=self.step - 1)
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/lib/python3.8/site-packages/meld/vault.py", line 498, in load_states
    positions = self.load_positions(stage)
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-meldmd-0.6.1-ylfrk2dho5jkzme6yvep2llyiw7bvbwe/lib/python3.8/site-packages/meld/vault.py", line 267, in load_positions
    return self._cdf_data_set.variables["positions"][..., stage]
  File "netCDF4/_netCDF4.pyx", line 4446, in netCDF4._netCDF4.Variable.__getitem__
  File "netCDF4/_netCDF4.pyx", line 4511, in netCDF4._netCDF4.Variable._toma
  File "/home/bsmits/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/py-numpy-1.24.1-qq2qr326fyt6pafzzj3opt5cg6irtodn/lib/python3.8/site-packages/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'bool'
```